doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+
+##### VxlanOrch processing
+
+- Create Tunnel if this is the first EVPN route to the remote VTEP. 
+- Retrieve the bridgeport associated with the dynamic tunnel.
+- create VlanMember of type tunnel and associate it to the vlanPort object in portsorch.
+- call sai_vlan_api->create_member with vlanid and tunnel bridgeport. 
+
+
+#### IMET route origination
+
+The IMET route origination sequence is depicted in the diagram below. 
+
+![IMET Origination](images/imetorig.PNG "Figure : IMET Route Origination")
+__Figure 7: IMET route origination__
+
+Zebra triggers BGP to originate IMET route when the vni device is programmed in the kernel.
+
+##### kernel programming vni device
+
+Following Linux operations are performed by vxlanmgr when VXLAN_TUNNEL_MAP is configured with VNI 1000 mapped to VLAN 100:
+
+```
+ip link add vxlan-100 type vxlan id 1000 local <src_vtep_ip> nolearning dstport 4789
+
+# Add VXLAN interface to Bridge and make it member of corresponding Vlan
+ip link set dev vxlan-100 master Bridge
+bridge vlan add vid 100 dev vxlan-1000
+bridge vlan add vid 100 untagged pvid dev vxlan-100
+
+ip link set up vxlan-100
+```
+
+
+
+Note that in case of L3VNI, FRR requires VRF to be configured with corresponding L3VNI, and also L3VNI is bound to VRF in Linux. Below Linux operation is performed by intfmgr when interface is bound to VRF.
+
+```
+ip link set dev Vlan100 master <vrf>
+```
+
+vni netdev creation will trigger the IMET route generation. 
+
+Since this programming is dependent on VLAN to be present a check is being added as part of the CLI to ensure that VLAN is present before the map is configured.
+
+The existing kernel programming sequence in VxlanMgr for tunnels triggered by CFG_VNET_TABLE entries is not altered.
+
+##### FRR processing
+
+VLAN creation and a VLAN-VNI mapping in Linux triggers origination of IMET route in BGP.  
+FRR originates IMET routes based on kernel events listened to by zebra. 
+
+##### VxlanOrch processing
+
+In the current implementation, kernel programming for VXLAN devices is done by vxlanmgr only when CFG_VNET_TABLE entry is made. 
+
+In this enhancement for BGP EVPN tunnels, VXLAN_TUNNEL_MAP entry addition along with presence of VLAN will trigger the kernel programming. 
+
+
+### 4.3.7 MAC route handling
+
+
+MAC addresses learnt in the forwarding plane are populated by fdborch in the FDB_TABLE in STATE_DB. The Linux FDB learns only those MAC addresses which are CPU bound (e.g. from ARP request/response). In order to provide EVPN L2 service, all of the MAC addresses learnt in the forwarding plane need to be advertised.
+
+In order to achieve above, new process fdbsyncd is added in SwSS. Fdbsyncd would subscribe to FDB entries for a given VLAN once VLAN-VNI mapping is available. Fdbsyncd will maintain the VLAN subscription and send FDB events to Linux FDB.
+
+![EVPN MAC route handling](images/l2macflowcombined.PNG )
+__Figure 8: EVPN MAC route handling__
+
+
```

Comment:
Any performance concern here to use kernel netlink message to parse all MAC?  It's well known that when the number of routes in network becomes large,  the number of kernel notification messages becomes a bottleneck and would be unreliable, and kernel expecting applications to handle that.

Reply:
We have not seeing much issue with 40K MAC handled from Kernel, We have tested on AS7326(TD3).